### PR TITLE
Add styling with active URLs and stop button

### PR DESCRIPTION
**Pull request by manzt**
Saw your message on Zulip.... Wish I'd pushed this sooner.

Screen Shot 2020-12-05 at 1 11 29 PM

### CODE REVIEW

**Comment by manzt** [comment]
I think we used the same template, ha! I can checkout main and add some of the route specific styling if you'd like. Maybe there are some ideas here that you'd like to apply.

File: templates/base.html [hunk 1]
```diff
@@ -7,24 +7,33 @@
...
+				<div class="flex-shrink-0">
+					<img class="h-8 w-8" src="https://github.com/napari/napari/raw/master/napari/resources/logo.png" alt="Workflow">
+				</div>
+				<div class="ml-10 flex items-baseline space-x-4">
+					{% for route in routes %}
+						{% if route.active %}
+						<a href="{{route.href}}" class="bg-gray-900 text-white px-3 py-2 rounded-md text-sm font-medium">{{route.name}}</a>
+						{% else %}
+						<a href="{{route.href}}" class="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium">{{route.name}}</a>
+						{% endif %}
```
Comment (manzt) [inline templates/base.html:23]: Applies different styling for "active" page.

**Comment by pwinston** [comment]
Yeah ha.

Feel free to add more. I changed the viewer WebGL part to no longer be full screen. That's worth keeping. Then we can put tailwind UI around it instead of just using the WebGL for GUI. 

Yeah I noticed my menu didn't highlight the active page. But figured it was something we could tack on.

**Comment by pwinston** [comment]
What does your stop button do?

Would be great to show whether the connection to webmon was active. And then separately is the connection to napari active. 

Typically if you are connected to webmon it's connected to napari. 

The exception is if you launched webmon alone. Which is fairly useless today, but I could see that changing. 

For instance, have a way to start napari from webmon. Inverting the usual order. 

Why? My thinking is you could start napari with a config you specified in webmon. For example webmon could display a list of URL's for remote datasets:

1. Run webmon.py
2. It opens Chrome to show webmon UI
3. Pick remote dataset from list
4. Click **Start Napari**.

Totally alternate use of webmon. Pure speculation if this would be useful. But I suspect it could evolve into something interesting. 

WebUI -> napari -> WebUI

**Comment by manzt** [comment]
> What does your stop button do?

I just added a link for: 

```python
@app.route("/stop")
def stop():
    """Stop the socketio server.

    The documentation says socketio.stop() "must be called from a HTTP or
    SocketIO handler function". So we have this endpoint which our
    on_shutdown() function hits. When socketio is stopped the
    socketio.run() call in main will return and the process will exit.
    """
    LOGGER.info("/stop -> stopping socketio.")
    socketio.stop()
    return "<h1>Stop</h1>Stopped socketio."
```

**Comment by manzt** [comment]
> Would be great to show whether the connection to webmon was active. And then separately is the connection to napari active.

Is there some socket message that is triggered when napari is active?

**Comment by pwinston** [comment]
I don't think there is yet. There are several connection messages related to JS->Webmon which are from the original demo, that we don't really use. What I see those being useful for is bringing the WebUI up to speed if it's connecting to an already running napari. For instance right now the Viewer will be wrong till you move the camera around some. And the Loder will have no stats till you load.

Both of those could benefit form some sort of "send me the current state" message or something.

I might plumb in the webmon start/stop and napari start/stop. Then we can worry about the UI for them.

But we'll see, I want to lay a good foundation, but not go overboard if no one uses webmon, or not for a long time, then a lot of this will be wasted.

But I know from my use what I'd like to see if I were using webmon daily. Which is something I could see certain developers doing once it has enough useful things in it.

**Comment by pwinston** [comment]
@manzt the route/pages stuff is awesome, very easy to add more, and showing the active page looks nice.

I had to add my hack "height: 800" style back to `viewer.html` or the webgl window was only like 100px tall for some reason. I might have messed up something with the merge. 

I think the "stop" button is a good step zero towards some of the stuff I'm talk about above. Am I connected to webmon? Is it connected to napari? And eventually maybe even starting napari.

I may or may not add some of that. My real goal is just 100% focus on only adding stuff I need for octree debugging. But it's tempting to add more b/c this is so greenfield, a lot to setup.

**Pull request merged**